src/sentry/static/sentry/app/components/alertMessage.jsx

```diff
@@ -5,23 +5,28 @@ import {t} from '../locale';
 
 const AlertMessage = React.createClass({
   propTypes: {
-    className: React.PropTypes.string,
-    id: React.PropTypes.number.isRequired,
-    message: React.PropTypes.string.isRequired,
-    type: React.PropTypes.string,
-    url: React.PropTypes.string
+    alert: React.PropTypes.shape({
+      id: React.PropTypes.string,
+      message: React.PropTypes.string.isRequired,
+      type: React.PropTypes.oneOf([
+        'success',
+        'error',
+        'warning'
+      ]),
+      url: React.PropTypes.string
+    })
```

Comment:
Also, this props change was mainly motivated by the fact that the `alert` itself is passed to `AlertActions.closeAlert`. This change happened since alerts are no longer guaranteed to have ID attributes, and it's assumed that it can be removed by reference from the `AlertStore`. This may be a dangerous assumption. I don't know.

An alternative option would be to flatten the props out again and use the `key` attribute to remove the alert from the `AlertStore`. But `key` is a reserved word and can't be used as `props`. :disappointed: 

If anybody has a strong opinion, I'm willing to change it — but to me both seem like less than ideal solutions.